### PR TITLE
test-spec: matter: update files in CI-matter-test

### PR DESCRIPTION
**Pull request by doublemis1**
Signed-off-by: Michał Szablowski <michal.szablowski@nordicsemi.no>

### CODE REVIEW

**Review by LuDuda — APPROVED** [review]

File: .github/test-spec.yml [hunk 1]
```diff
@@ -84,7 +84,20 @@
...
-  - "**/*"
+  - "include/dfu/**/*"
+  - "include/mgmt/mcumgr/**/*"
+  - "soc/arm/nordic_nrf/**/*"
+  - "subsys/dfu/**/*"
+  - "subsys/settings/**/*"
+  - "subsys/net/**/*"
+  - "subsys/mgmt/mcumgr/**/*"
+  - "drivers/net/**/*"
+  - "drivers/wifi/**/*"
```
Comment (Damian-Nordic) [inline .github/test-spec.yml:95]: nRF700X drivers are in sdk-nrf, not zephyr, so this is probably not needed.

File: .github/test-spec.yml [hunk 2]
```diff
@@ -84,7 +84,20 @@
...
+  - "samples/hci_rpmsg/**/*"
```
Comment (Damian-Nordic) [inline .github/test-spec.yml:96]: Are these samples entries needed? Do our tests depend on mcumgr samples?

File: .github/test-spec.yml [hunk 3]
```diff
@@ -84,7 +84,20 @@
...
+  - "samples/subsys/mgmt/mcumgr/smp_svr/**/*"
+  - any:
+    - "subsys/bluetooth/**/*"
```
Comment (Damian-Nordic) [inline .github/test-spec.yml:99]: probably only `bluetooth/host` and `bluetooth/controller` are needed? At least `bluetooth/audio` shouldn't trigger our tests.

**Review by Damian-Nordic — APPROVED** [review]

File: .github/test-spec.yml (hunk 2 above)
Comment (doublemis1) [reply .github/test-spec.yml:96]: Do you mean hci_rpmsg?
We use that in wifi sample.

Regarding next record mcumgr, I thought we were basing our implementation of SMP support on this example

Comment (Damian-Nordic) [reply .github/test-spec.yml:95]: We use SMP server, but not the sample, so this shouldn't affect us, I think.

Comment (doublemis1) [reply .github/test-spec.yml:95]: ok, removed

**Review by thst-nordic — APPROVED** [review]

**Pull request merged**